components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -815,4 +816,30 @@ private String getIdpSessionKey(String accessToken) {
         }
         return idpSessionKey;
     }
+
+    /**
+     * Check the requirement of having a configured public certificate or JWKS in SP and throw an exception with an
+     * error message if the public certificate or JWKS in SP is not configured.
+     *
+     * @param clientId     Client ID of the service provider.
+     * @param tenantDomain Tenant domain of the service provider.
+     * @throws IdentityOAuth2Exception Error when a JWKS endpoint or the certificate is not configured.
+     */
+    private void checkIfPublicCertConfiguredForEncryption(String clientId, String tenantDomain)
+            throws IdentityOAuth2Exception {
+
+        try {
+            if (StringUtils.isBlank(OAuth2Util.getSPJwksUrl(clientId, tenantDomain))) {
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("Jwks uri is not configured for the service provider associated with " +
+                            "client_id: %s , Checking for x509 certificate.", clientId));
+                }
+                OAuth2Util.getX509CertOfOAuthApp(clientId, tenantDomain);
+            }
+        } catch (IdentityOAuth2Exception e) {
+            throw new IdentityOAuth2Exception("You have enabled ID token encryption without configuring a " +
```

Comment:
Add clientId and tenant domain in the log. Otherwise would be hard to track which app failed.
Also consider the consistency of the format(or rather the message) of the log with the rest of the error logs

Reply:
Addressed with 3339067bc566fe9b9f6fb109d966334fad9a0bd2